src/query/interpret/eval.hpp

```diff
@@ -847,6 +849,9 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
       }
       auto res = function.function_(arguments, function.arguments_.size(), function_ctx);
       MG_ASSERT(res.GetMemoryResource() == ctx_->memory);
+      if (!is_transactional && res.ContainsDeleted()) {
+        return TypedValue(ctx_->memory);
```

Comment:
One general question, where was exception thrown before if object was deleted, where is this check?

Reply:
I think it was thrown by the client